src/input/ProcessInput.java

```diff
@@ -0,0 +1,124 @@
+package input;
+import assets.Flower;
+import assets.Peashooter;
+import engine.Game;
+import engine.Purse;
+import levels.LevelInfo;
+import view.Board;
+
+public class ProcessInput 
+{
+	private LevelInfo lvl;
+	private Game game;
+	private Board board;
+	private Purse purse;
+	
+	public ProcessInput(LevelInfo lvl, Game game, Board board, Purse purse)
+	{
+		this.lvl = lvl;
+		this.game = game;
+		this.board = board;
+		this.purse = purse;
+	}
+	
+	public boolean processCommand(Command command) //return false if cmd is processCommand is unsuccessful
+	 {
+	        if(command.isUnknown())
+           {
+	            System.out.println("I don't know what you mean...");
+	            return false;
+	        }
+	        
+	        String commandWord = command.getCommandWord();
+	        if (commandWord.equals("help")) 
+	        {
+	            printHelp();
+	            return true;
+	        }
+	        if (commandWord.equals("load") && command.getSecondWord() != null)
+	        {
+	            System.out.println("This feature is not avaliable yet");
+	            return true;
+	        }
+	        else if (commandWord.equals("play")) 
+	        {
+	        	if(command.getSecondWord() != null)
+	        	{
+	        		game.play(Integer.valueOf(command.getSecondWord()));  //where commandSecondWord is the level selector
+	        	}
+	        	
+	        	game.play(null);
+	            
+	            return true;
+	        }
+	        else if (commandWord.equals("quit")) 
+	        {
+	           game.setFinished();
+	           System.out.println("User quit the game");
+	           return true;
+	        }
+	        else if (commandWord.equals("pass"))
+	        {
+	        	return true;
+	        }
+	        
+	        
+	        
+	        //======================================================== placing units commands
+	        
+	        else if(command.getCommandWord().equals("place"))
+	        {
+	        	if(command.getThirdWord() == null || command.getFourthWord() == null)             //restriction 1
+	            {
+		            System.out.println("I don't know what you mean...you forgot to place coordinates");
+		            return false;
+		        }
+		        else if(!inRange(Integer.valueOf(command.getThirdWord()), Integer.valueOf(command.getFourthWord())))  //restriction 2  see if the coordinates are in range
+		        {
+		        	 System.out.println("The inputted coordinates are out of range");
+			         return false;
+		        }
+	        	
+	        	
+	        	if (command.getSecondWord().equals("sf") && purse.redeemPoints(10) && lvl.isValidUnit("sf"))
```

Comment:
This will need to be reworked once the units are more fleshed out. It will also need to be reworked for LevelInfo. But for now, we can leave it.